assignment_3/p1.cpp

```diff
@@ -0,0 +1,231 @@
+//CS101 PROJECT1
+#include <iostream>
+#include <string>
+#include <fstream>
+#include <sstream>
+#include <vector>
+#include <cctype>
+#include <iomanip>
+#include <limits>
+
+
+using namespace std;
+
+void printHelp();
+void printColumns (const vector<string>& columns, int colWidth);
+void printRows (const vector<vector<string> >& rows, int colWidth);
+void search (const vector<vector<string> >& rows, const vector<string>& columns, const string& colName, const string& value, int colWidth);
+void findMinMaxAvg (const vector<vector<string> >& rows, const vector<string>& columns, const string& colName, const string& operation);
+
```

Comment:
I do not like the signatures of the search or Find MinMaxAvg subroutines. I think it's appropriate that the subroutines that are solely meant to print to the command line are void. However these two subroutines (IMO) should return data. I'd suggest something like:

vector<std::vector<string>> search(vector<vector<string> rows, vector<string> columns, string colName, string operation)

AND

double findMinMaxAvg(vector<vector<string> rows, vector<string> columns, string colName, string operation)